packages/lesswrong/components/users/DialogueMatchingPage.tsx

```diff
@@ -479,262 +1013,250 @@ const DialogueCheckBox: React.FC<{
-    fragmentName: "DialogueCheckInfo",
-    collectionName: "DialogueChecks",
-  });
-
-  const {loading: userOptedInLoading, results: usersOptedInToDialogueFacilitation, loadMoreProps} = useMulti({
-    terms: { 
-      view: 'usersWithOptedInToDialogueFacilitation',
-      limit: 10, 
-        },
-    fragmentName: 'UsersOptedInToDialogueFacilitation',
-    collectionName: 'Users'  
-  });
+  if (!currentUser) return <p>You have to be logged in to view this page</p>
 
   if (loading) {
     return <Loading />;
   } else if (!usersOptedInToDialogueFacilitation) {
     return <p>Error...</p>;
   }
 
-  const targetUserIds = userDialogueChecks?.map(check => check.targetUserId) ?? [];
-
-  async function createDialogue(title: string, participants: string[]) {
-    const createResult = await createPost({
-      data: {
-        title,
-        draft: true,
-        collabEditorDialogue: true,
-        coauthorStatuses: participants.map(userId => ({userId, confirmed: true, requested: false})),
-        shareWithUsers: participants,
-        sharingSettings: {
-          anyoneWithLinkCan: "none",
-          explicitlySharedUsersCan: "edit",
-        },
-        contents: {
-          originalContents: {
-            type: "ckEditorMarkup",
-            data: ""
-          }
-        } as AnyBecauseHard
-      },
-    });
-    if (createResult?.data?.createPost?.data) {
-      const post = createResult?.data?.createPost?.data;
-      if (post) {
-        const postId = post._id;
-        const postEditUrl = `/editPost?postId=${postId}`;
-        navigate(postEditUrl);
-      }
-    }
-  }
+  const userDialogueUsefulData: UserDialogueUsefulData = data?.GetUserDialogueUsefulData;
 
-  if (!currentUser) return <p>You have to be logged in to view this page</p>
+  const matchedUsers: UsersOptedInToDialogueFacilitation[] | undefined = matchedUsersResult?.GetDialogueMatchedUsers;
+  const matchedUserIds = matchedUsers?.map(user => user._id) ?? [];
+  const topUsers = userDialogueUsefulData?.topUsers.filter(user => !matchedUserIds.includes(user._id));
+  const recentlyActiveTopUsers = topUsers.filter(user => user.recently_active_matchmaking)
+  const nonRecentlyActiveTopUsers = topUsers.filter(user => !user.recently_active_matchmaking)
+  const dialogueUsers = userDialogueUsefulData?.dialogueUsers.filter(user => !matchedUserIds.includes(user._id));
+  const optedInUsers = usersOptedInToDialogueFacilitation.filter(user => !matchedUserIds.includes(user._id));
+  const activeDialogueMatchSeekers = userDialogueUsefulData?.activeDialogueMatchSeekers.filter(user => !matchedUserIds.includes(user._id));
+  
   if (loading) return <Loading />
-  if (error || !userDialogueChecks || userDialogueChecks.length > 1000) return <p>Error </p>; // if the user has clicked that much stuff things might break...... 
+  if (error ?? !userDialogueChecks ?? userDialogueChecks.length > 1000) return <p>Error </p>; // if the user has clicked that much stuff things might break...... 
```

Comment:
I think maybe there was a misunderstanding about `??` vs `||`... you do in fact still need to use `||` if you're trying to perform a logical OR operation ("return true if any of these conditions are true").

Reply:
yeah I overdid it